File: C#/IsValid.cs
public class Solution {
    public bool IsValid(string s) {
        
        // Default
        if (s.Length % 2 == 1)
        {
            return false;
        }
        
        List<char> open = new List<char>();
        
        for (int i = 0; i < s.Length; i++)
        {
            if (s[i] == '(' || s[i] == '[' || s[i] == '{')
            {
                open.Add(s[i]);
            }
            else
            {   
                if (open.Count == 0)
                {
                    return false;
                }
                
                if (s[i] == ')' && open[open.Count - 1] != '(')
                {
                    return false;
                }
                else if (s[i] == ']' && open[open.Count - 1] != '[')
                {
                    return false;
                }
                else if (s[i] == '}' && open[open.Count - 1] != '{')
                {
                    return false;
                }
                else
                {
                    open.RemoveAt(open.Count - 1);
                }
            }
        }
        
        if (open.Count != 0)
        {
            return false;
        }
        else
        {
            return true;
        }
    }    
    
}

File: Python/isValid.py
class Solution:
    def isValid(self, s: str) -> bool:

        while (len(s) > 0):
            if(s.find("()") != -1):
                s = s.replace("()","")

            elif(s.find("{}") != -1):
                s = s.replace("{}","")

            elif(s.find("[]") != -1):
                s = s.replace("[]","")
            else:
                break

        return (s == "")
        